lib/modules/datasource/docker/index.ts

```diff
@@ -558,6 +594,68 @@ export class DockerDatasource extends Datasource {
     return null;
   }
 
+  @cache({
+    namespace: 'datasource-docker-architecture',
+    key: (
+      registryHost: string,
+      dockerRepository: string,
+      currentDigest: string
+    ) => `${registryHost}:${dockerRepository}@${currentDigest}`,
+    ttlMinutes: 1440 * 28,
+  })
+  public async getImageArchitecture(
+    registryHost: string,
+    dockerRepository: string,
+    currentDigest: string
+  ): Promise<string | null | undefined> {
+    try {
+      const manifestResponse = await this.getManifestResponse(
+        registryHost,
+        dockerRepository,
+        currentDigest,
+        'head'
+      );
+
+      if (
+        manifestResponse?.headers['content-type'] !== MediaType.manifestV2 &&
+        manifestResponse?.headers['content-type'] !== MediaType.ociManifestV1
+      ) {
+        return null;
+      }
+
+      const configDigest = await this.getConfigDigest(
+        registryHost,
+        dockerRepository,
+        currentDigest
+      );
+      if (!configDigest) {
+        return null;
+      }
+
+      const configResponse = await this.getImageConfig(
+        registryHost,
+        dockerRepository,
+        configDigest
+      );
+      if (configResponse) {
+        const architecture = configResponse.body.architecture ?? null;
+        logger.debug(
+          `Current digest ${currentDigest} relates to architecture ${
+            architecture ?? 'null'
+          }`
+        );
+
+        return architecture;
+      }
+    } catch (err) /* istanbul ignore next */ {
+      if (err.statusCode !== 404) {
+        throw err;
+      }
+    }
```

Comment:
```suggestion
    }
    logger.debug({ err, currentDigest, dockerRepository, registryHost }, 'Docker image config not found');
```

Reply:
Added back the msg I had until https://github.com/renovatebot/renovate/pull/16554#discussion_r929573124 and  https://github.com/renovatebot/renovate/pull/16554/commits/ec37c818530196ddec344655459e867f0d3886a8#diff-3ab04cc849530c4050b0c8705a0a6837b99a76c8ea57a33a2f061fee5a7fdc4c